multimodal/src/autogluon/multimodal/utils/export.py

```diff
@@ -160,23 +152,85 @@ def export_onnx(
             else:
                 batch[key] = inp.to(device)
         self._model.to(device)
-        input_vec = InputBatch(**batch)
+        keys_to_drop = list(set(batch.keys()).difference(set(self._model.input_keys)))
+        for k in keys_to_drop:
+            batch.pop(k)
+        # Truncate input data types for TensorRT (only support: bool, int32, half, float)
+        if truncate_long_and_double:
+            for k in batch:
+                if batch[k].dtype == torch.int64:
+                    batch[k] = batch[k].to(torch.int32)
+        input_vec = [batch[k] for k in self._model.input_keys]
```

Comment:
Could we move all the logic to process/convert `batch` data, including the code to support mixed precision on floating point inputs, to util function `get_processed_batch_for_deployment `?  I think these logic better fits that context.

Reply:
moved to get_processed_batch_for_deployment function as suggested.